taskhandler

susisstrolch opened the pull request:
simple syslog client
syslog(uint8_t facility, uint8_t severity, const char tag[], const char message[], ...)

send a UDP syslog message to a syslog server.
current config parameters (on html home page):
- SNTP server
- Syslog server

pull request contains the required task handler.

tve committed:
reduce size of icons

tve committed:
oops, readd mistakenly deleted file

tve committed:
make esp-12 with 4MB flash work

tve committed:
clear bssid before reassociating

Commit:
merge upstream:master

Commit:
conditionaly include "local.conf" from Makefile

for convenience reasons included from Makefile, added to gitignore..

Commit:
merge from origin

Commit:
simple handler for system_os_task/system_os_post

we wrap both functions in much easier ones (register_usr_task, post_usr_task).
register_usr_task requires an os_task_t parameter (the task itself) and returns a
task number (0...MAXUSRTASKS) which is later on used by "post_usr_task".
register_usr_task is self-initializing. no static arrays - malloc only...

Commit:
syslog - a simple syslog client

Commit:
syslog - a simple syslog client

syslog UDP messages are send via
syslog(uint8_t facility, uint8_t severity, const char tag[], const char message[], ...);

the syslog function queues any msg on system heap. If heap reaches a minimum size (8kB),
syslog stops queuing.
As soon as Wifi connectivity is available, syslog will send out all queued messages.
Queuing will be reenabled after all pending messages are sent.

Commit:
fix rebase / merge errors

- missing LOG_xxx macros in syslog.h
- fixed tag (esp-link) in LOG_xxx macro

tve commented on esp-link/config.c:
```diff
@@ -9,21 +9,31 @@
+  .reset_pin    = MCU_RESET_PIN, .isp_pin      = MCU_ISP_PIN,
+  .conn_led_pin = LED_CONN_PIN,  .ser_led_pin  = LED_SERIAL_PIN,
+  .baud_rate    = 115200,
+  .hostname     = "ems-link02\0",
```
I don't think this makes sense as a default

susisstrolch replied:
You mean the designated initializer per se or my (accidently promoted) 
hostname?

On 11/24/2015 06:45 AM, Thorsten von Eicken wrote:

> In esp-link/config.c 
> https://github.com/jeelabs/esp-link/pull/73#discussion_r45698825:
> 
> > -  0, 0x00ffffff, 0,                    // static ip, netmask, gateway
> > -  0,                                   // log mode
> > -  0,                                   // swap uart (don't by default)
> > -  1, 0,                                // tcp_enable, rssi_enable
> > -  "\0",                                // api_key
> > -  0, 0, 0,                             // slip_enable, mqtt_enable, mqtt_status_enable
> > -  2, 1,                                // mqtt_timeout, mqtt_clean_session
> > -  1883, 60,                            // mqtt port, mqtt_keepalive
> > -  "\0", "\0", "\0", "\0", "\0",        // mqtt host, client_id, user, password, status-topic
> > -  "\0",                                // system description
> > -  1,                                   // rx_pullup
> > -  .seq=33, .magic=0, .crc=0,
> > -  .reset_pin    = MCU_RESET_PIN, .isp_pin      = MCU_ISP_PIN,
> > -  .conn_led_pin = LED_CONN_PIN,  .ser_led_pin  = LED_SERIAL_PIN,
> > -  .baud_rate    = 115200,
> > -  .hostname     = "ems-link02\0",
> 
> I don't think this makes sense as a default
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/jeelabs/esp-link/pull/73/files#r45698825.

tve replied:
I mean the hostnames. I like the way you changed the initializers, didn't know one could do that :-).

tve commented on esp-link/config.c:
```diff
@@ -9,21 +9,31 @@
+  .sys_descr 	= "\0",
+  .rx_pullup	= 1,
+  .sntp_server  = "diskstation\0",
+  .syslog_host  = "diskstation\0",
```
Mhh, not sure this is a reasonable default...

tve commented on esp-link/config.c:
```diff
@@ -9,21 +9,31 @@
+  .rx_pullup	= 1,
+  .sntp_server  = "diskstation\0",
+  .syslog_host  = "diskstation\0",
+  .syslog_minheap= 8192,
```
Whoa, that's a lot!

susisstrolch replied:
It's not the size of the syslog buffer - it's the required overall heapsize when (trying to) queue a syslog message.

tve replied:
Oh, I misunderstood then, sorry. I have not reviewed the syslog.c file yet... Does this have to be a configurable value instead of just a constant in the code? How would the user know what a reasonable value is? Also, why 8KB as opposed to less?

tve commented on syslog/syslog.c:
```diff
@@ -0,0 +1,375 @@
+    } else {
+    //re-arm timer to check ip
+    os_timer_setfn(&check_udp_timer, (os_timer_func_t *)syslog_check_udp, NULL);
+    os_timer_arm(&check_udp_timer, 100, 0);
```
Does it make sense to check every 100ms? Seems like checking every 1 or even 10 seconds would be fine? Maybe it doesn't matter...

susisstrolch replied:
Usually (at least with my router) Wifi is up after approx. 2.5s. So, a 
10s increment would be a bit rough.
The 1s sounds like a compromize.

On 11/24/2015 07:01 AM, Thorsten von Eicken wrote:

> In syslog/syslog.c 
> https://github.com/jeelabs/esp-link/pull/73#discussion_r45699483:
> 
> > -    if (wifi_station_get_connect_status() == STATION_GOT_IP && ipconfig.ip.addr != 0)
> > -    {
> > -    if (syslogState == SYSLOG_WAIT) {          // waiting for initialization
> > -    DBG_SYSLOG("connected, initializing UDP socket\n");
> > -    syslog_init(flashConfig.syslog_host);
> > -    }
> > -    } else {
> > -    DBG_SYSLOG("waiting 100ms\n");
> > -    if ((wifi_station_get_connect_status() == STATION_WRONG_PASSWORD ||
> > -    wifi_station_get_connect_status() == STATION_NO_AP_FOUND ||
> > -    wifi_station_get_connect_status() == STATION_CONNECT_FAIL)) {
> > -    os_printf("**\* connect failure!!! \r\n");
> > -    } else {
> > -    //re-arm timer to check ip
> > -    os_timer_setfn(&check_udp_timer, (os_timer_func_t *)syslog_check_udp, NULL);
> > -    os_timer_arm(&check_udp_timer, 100, 0);
> 
> Does it make sense to check every 100ms? Seems like checking every 1 
> or even 10 seconds would be fine? Maybe it doesn't matter...
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/jeelabs/esp-link/pull/73/files#r45699483.

Commit:
fixes/changes after pull request

- changed defaults for Flashconfig
  - SNTP-Server: empty
  - Syslog-Server: empty
  - hostname: reverted to esp-link

- heavy cleanup in syslog.c
  - removed debug messages
  - add local syslog_compose function

Commit:
espmissingincludes.h

encapsulate os_printf macro with do/while to avoid
unwanted side effects during macro expansion
